src/client-side-encryption/client_encryption.ts

```diff
@@ -948,12 +956,13 @@ export interface ClientEncryptionRewrapManyDataKeyResult {
 /**
  * @public
  * RangeOptions specifies index options for a Queryable Encryption field supporting "rangePreview" queries.
- * min, max, sparsity, and range must match the values set in the encryptedFields of the destination collection.
+ * min, max, sparsity, trimFactor and range must match the values set in the encryptedFields of the destination collection.
  * For double and decimal128, min/max/precision must all be set, or all be unset.
  */
 export interface RangeOptions {
   min?: any;
   max?: any;
-  sparsity: Long;
+  sparsity?: Long;
+  trimFactor?: Int32;
```

Comment:
Why is this strictly the Int32 type, couldn't `number` also be allowed? Same for sparsity and `bigint` right?

Reply:
Yeah.  I chose to use the exact BSON types because 1) the spec is particular about the exact BSON types, so we should be as well. Ensuring that users take the extra few characters to explicitly provide the BSON type we expect seems fine.

Reply:
Arguably fair for `number` but it's more than a few characters, its an object vs a number, it won't respond to `typeof` or other operations the same way a number would. 

I suggest we consider validating, wraping in an Int32 or, similar to how we treat server-side values, allow it to passthrough and fail as an incorrect type. In general, we define intuitive and idiomatic APIs for developing applications with MongoDB, so we want to meet users where they are rather than requiring them to write code specific to our domain (unless necessary). 

However, for bigint, I don't think the same inexactness applies, it can only ever be an int64

I imagine these values work without modification, the question is whether our Typescript is telling the truth about the reality of what works or what we chose to support, is that correct?

Reply:
Ensuring users provide the correct type helps both us and users - we can be sure that the user has provided the type we expect, and users see a self-documenting API that doesn't let them provide incorrect data.  Even if primitives do work, it's a more explicit and clear user experience if we ensure that users provide us the type of data we expect.

Reply:
The best documentation is actual documentation, if this type concern is critical, we should add some prose that informs them. Prose can be translated and disseminated more broadly than a compiler error that _only implies_ something about a part of the functionality relating to this encryption API.

For the `number` scenario, I disagree with introducing artificial limitations when code already automatically supports a broader set of inputs that can be tested and maintained by passing both types, it would only be one more case to cover.

For the `bigint` scenario, what is unclear, unspecific, or unexpected about that input?

Reply:
> The best documentation is actual documentation, if this type concern is critical, we should add some prose that informs them. Prose can be translated and disseminated more broadly than a compiler error that only implies something about a part of the functionality relating to this encryption API.

I fundamentally disagree with this.  Types are documentation and they are inherently superior to prose/doc comments.  Prose easily gets stale and out-of-date - types are self documenting.  I don't just mean compiler errors - when a user sees a function signature with concrete types in it, they know exactly what they're expected to provide.  In this scenario, an `Int32` clearly communicates that the type is numeric, integral and has to be inside the range of a signed, 32bit integer.  

Furthermore, w.r.t. number and bigint, we would need to provide additional doc comments and test cases.  Numbers can have decimal places or be out of range of int32.  Bigints can be greater than Longs.  All of this would require explicit documentation and potentially additional test cases.

Reply:
Prose documentation is not instead of accurate types we want to have and maintain both because they help us as well as our users. Explicit documentation should be added regardless of the input type.

We do not ask for the `Int32` boxed type anywhere else in the driver despite there being a number of options that can also only be within this bit size. `bigint` does not share the same problem that `numbers` do with respect to how they are handled in BSON, while the bigint may be larger, it can never be serialized to more or less than a 64 bit number. 

I found what I think is a [comparable example](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/crypt/explicit_encryption_context.rb#L139-L143) ruby changes the input to a boxed type, maybe it throws, maybe it coerces, but they're aligning with the common driver goal of providing an idiomatic API for their users.

Reply:
I feel using the BSON types here is appropriate as a means of making the API more specific and explicitly spec-compliant, but regardless of the option we go with the types, we should still have some prose documentation to specify the valid range of values. e.g. can the `trimFactor` be negative, or is it limited to zero and positive integers?